### PR TITLE
tsh scp to use target directory correctly

### LINKED ISSUE #5497: tsh scp does not respect the target directory
### Description

**What happened**:
When copying files with `tsh scp -r dir/ user@node:/dest/dir`, the files are always copied to the user's home directory on `node`.

**What you expected to happen**:
The directory (incl. all files) should be copied to the corresponding target directory if specified.

`tsh scp -r dir user@node/dest/dir` should copy `dir/*` as `/dest/dir/dir/*` onto `node`.
`tsh scp -r dir/* user@node/dest/dir` should copy `dir/*` as `/dest/dir/*` onto `node`.

**How to reproduce it (as minimally and precisely as possible)**:

### Environment
- Teleport version (use `teleport version`): 6.0.0-alpha.2

### PR DESCRIPTION
**Pull request by a-palchikov**
Fixes the scp behavior error I introduced when working on adding support for preserving file times [here](https://github.com/gravitational/teleport/issues/2889).

This now takes target directory into account in sink mode as previously.

Also, while looking into https://github.com/gravitational/teleport/issues/3235 I noticed that the low disk space condition is not reported properly to the user. Now, the error is exposed to show what's failed.
Additionally, when copying big(ger) files, I noticed that interrupting the command did not work and only exited once the file copy was completed:

```
tsh scp -r files/1g.img root@teleport-node:/root/
^C
```

### Copying a large file to the node that's low on space:

Before:
```
$ tsh scp -r files/1g.img root@teleport-node:/root/files/
error: Process exited with status 1
```
With multiple files:
```
$ tsh scp -r files/* root@teleport-node:/root/files/
error: Process exited with status 1
```

After:
```
$ tsh scp  files/1g.img root@teleport-node:/root/files
error: write /root/files/1g.img: no space left on device
error: Process exited with status 1
```
With multiple files:
```
$ tsh scp files/* root@teleport-node:/root/files
error: write /root/files/1g.img: no space left on device
error: write /root/files/2g.img: no space left on device
error: write /root/files/test1.img: no space left on device
error: write /root/files/test2.img: no space left on device
error: write /root/files/test.img: no space left on device
error: Process exited with status 1
```

Fixes https://github.com/gravitational/teleport/issues/5497.
Updates https://github.com/gravitational/teleport/issues/3235.

### CODE REVIEW

File: lib/client/api.go [hunk 1]
```diff
@@ -1543,13 +1548,13 @@ func (tc *TeleportClient) SCP(ctx context.Context, args []string, port int, flag
 				return trace.Wrap(err)
 			}
 
-			err = client.ExecuteSCP(cmd)
+			err = client.ExecuteSCP(ctx, cmd)
 			if err != nil {
-				return onError(err)
+				exitErr = err
```
Comment (a-palchikov) [inline lib/client/api.go:1553]: Only saves the last exit error but it should be ok. Not breaking out of the loop for multiple files follows the existing behavior of OpenSSH `scp` - let me know if that's not desired.

Comment (fspmarshall) [reply lib/client/api.go:1569]: Perhaps the errors could be logged s.t. `tsh scp --debug` shows intermediate errors as well?  Potentially useful.

**Review by fspmarshall — APPROVED** [review]

File: lib/sshutils/scp/scp.go [hunk 2]
```diff
@@ -386,23 +389,25 @@ func (cmd *command) serveSink(ch io.ReadWriter) error {
...
-			return trace.Wrap(err)
-		}
-		if mode := fi.Mode(); !mode.IsDir() {
+		if !cmd.FileSystem.IsDir(cmd.Flags.Target[0]) {
 			return trace.BadParameter("target path must be a directory")
 		}
 	}
 
+	rootDir := localDir
+	if len(cmd.Flags.Target) != 0 && cmd.FileSystem.IsDir(cmd.Flags.Target[0]) {
```
Comment (r0mant) [inline lib/sshutils/scp/scp.go:401]: Could you add a comment here about what this condition checks and why.

File: lib/sshutils/scp/scp.go [hunk 3]
```diff
@@ -478,14 +483,13 @@ func (cmd *command) processCommand(ch io.ReadWriter, st *state, b byte, line str
...
 
-	// if the destination path is a folder, we should save the file to that folder, but
-	// only if 'recursive' is set
-
-	path := cmd.Flags.Target[0]
-	if cmd.Flags.Recursive || cmd.FileSystem.IsDir(path) {
-		path = st.makePath(fc.Name)
+	// Unless target specifies a file, use the file name from the command
+	filename := fc.Name
+	if !cmd.Flags.Recursive && !cmd.FileSystem.IsDir(cmd.Flags.Target[0]) {
```
Comment (r0mant) [inline lib/sshutils/scp/scp.go:488]: Is `cmd.Flags.Target` guaranteed to have at least 1 element here?

**Review by r0mant — APPROVED** [review]

**Review by russjones — APPROVED** [review]
Bot.

File: lib/sshutils/scp/scp.go (hunk 2 above)
Comment (a-palchikov) [reply lib/sshutils/scp/scp.go:401]: I'll move that to a self-explanatory helper

File: lib/sshutils/scp/scp.go (hunk 3 above)
Comment (a-palchikov) [reply lib/sshutils/scp/scp.go:488]: Yes, I've updated the client side of SCP to send files in bulk for source mode and the caller makes sure that at least 1 target item is always set.

File: lib/client/api.go (hunk 1 above)
Comment (a-palchikov) [reply lib/client/api.go]: I decided to roll back to old behavior of exiting upon first encountered error as it seems to make more sense.

**Pull request merged**